math/params.go

```diff
@@ -14,41 +14,42 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with goUmbral. If not, see <https://www.gnu.org/licenses/>.
-package umbral
+package math
 
-// #include "shim.h"
-import "C"
+import (
+    "github.com/nucypher/goUmbral/openssl"
+)
 
 type UmbralParameters struct {
-    Curve Curve
+    Curve *openssl.Curve
     Size uint
-    G Point
-    U Point
+    G *Point
+    U *Point
 }
 
-func GetNewUmbralParameters(curve Curve) (UmbralParameters, error) {
+func NewUmbralParameters(curve *openssl.Curve) (*UmbralParameters, error) {
     var params UmbralParameters
     params.Curve = curve
     params.Size = curve.FieldOrderSize()
 
     params.G = GetGeneratorFromCurve(curve)
     gBytes, err := params.G.ToBytes(true)
     if err != nil {
-        return UmbralParameters{}, err
+        return nil, err
     }
 
     parametersSeed := []byte("NuCypher/UmbralParameters/")
 
     parametersSeed = append(parametersSeed, byte('u'))
```

Comment:
I'm looking at pyUmbral's version here:
```
parameters_seed = b'NuCypher/UmbralParameters/'
self.u = unsafe_hash_to_point(g_bytes, self, parameters_seed + b'u')
```
It adds one additional 'u' at the end of the seed. I suppose I could have just included that 'u' at the end of the first string rather than appending it in a call to append.